js-intro.js

```diff
@@ -48,49 +68,96 @@ var myMessage = "Learning is fun!"
 // 3a. Write the code that logs each letter of the message using a for loop.
 //hint: "logs" in this case means "prints to the terminal console."  This is typically done with a console.log(x), where x is the string, function, etc that you wish to see in the terminal.
 
+const logEach = (string) => {
+    for (let i = 0; i < string.length; i++) {
+        console.log(string[i]) 
+    }
+}
+
+logEach(myMessage)
 
 // 3b. Write the code that logs each letter of the message using map.
 
+var newMessage = myMessage.split("")
+
+const logMap = (array) => {
+    array.map(letter => console.log(letter))
+}
 
+logMap(newMessage)
 
 // 3c. STRETCH: Write the code that logs each letter of the message using a while loop.
 
 
+const whileTest = (string) => {
+    var i = 0
+    while (i < string.length) {
+        console.log(string[i])
+        i++
+    }
+}
 
-// 3d. SUPER STRETCH: Write the code that logs each letter of the message using forEach().
+whileTest(myMessage)
 
+// 3d. SUPER STRETCH: Write the code that logs each letter of the message using forEach().
 
+const forEachTest = (array) => {
+    array.forEach(letter => console.log(letter))
+}
 
+forEachTest(newMessage)
 
 // ---------------------------- Consider the following variable:
 
 var testString = "thisisateststring"
 
 // 4a. Write a function that takes the variable as an argument and returns the string without vowels. Expected output: "thsstststrng"
 
-
-
-
+const noVowels = (string) => {
+
+    if (typeof string=== "string") {
+        let vowels = ["a", "e", "i", "o", "u"]
+        let array = string.split("")
+        
+        let answer = array.filter(value => {
+            for (let i = 0; i<value.length; i++) {
+                if (value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u"){
+                    return value
+                }
+            }
+        })
+        return answer.join("")
+    }   else {
+        return "Please enter a valid string"
+    }
+}
+
+console.log(noVowels(testString))
 
```

Comment:
Nice work handling edge cases. I don't think the vowels array is doing much for you. The rest of your code is taking care of what is needed.